src/Universum/Catstraints.hs

```diff
@@ -0,0 +1,79 @@
+{-# LANGUAGE ConstraintKinds  #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE QuasiQuotes      #-}
+{-# LANGUAGE Rank2Types       #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeOperators    #-}
+
+-- | Provides means to envade your system with kots.
+
+module Universum.Catstraints where
+
+import           Control.Concurrent (threadDelay)
+import           Control.Monad      (forever)
+import           Control.Monad.Fix  (fix)
+import           Data.Reflection
+import           Data.Text
+import           Debug.Trace
+import           GHC.Conc.Sync      (par)
+import           GHC.IO.Unsafe      (unsafePerformIO)
+import           Prelude
+import           System.Process     (callProcess)
+import           Text.RawString.QQ  (r)
+
+-- | Power of kots is enourmous and should be properly controlled.
+data (/◕¤◕/) = VIAL_OF_CATNIP
+type (^-^/) = Given (/◕¤◕/)
+
+-- | Let the madness begin.
+unleashKots :: ((^-^/) => a) -> a
+unleashKots = give VIAL_OF_CATNIP
+
+-- | Yet quite harmless kot.
+(🐱) :: (^-^/) => a -> a
+(🐱) x = x
+
+-- | Silently spreads over your RAM until OOM.
+(～￣▽￣～) :: (^-^/) => a -> a
+(～￣▽￣～) = par (mconcat $ repeat @Text "kot!!! ")
+
+-- | This cat is watching on you and waiting for stroking.
+(＾*･.･*＾) :: (^-^/) => a -> a
+(＾*･.･*＾) = trace "pur\n" . (＾*･.･*＾)
```

Comment:
I think it's better to track `purr` here. It's not clear, what `pur` means. It might be related to `purescript` and we definitely don't want confusing here.